js/app.js

```diff
@@ -1,18 +1,37 @@
 import { firstOrDefaultValue, firstOrDefaultContent } from "/js/parsing.js";
 
 const projectId = "99b0b3b0-4838-0051-3d57-8af72f55e8a0";
-const apiItems = "https://deliver.kontent.ai/{0}/items{1}";
-const apiItem = "https://deliver.kontent.ai/{0}/items/{1}";
+const apiItems = `https://deliver.kontent.ai/${projectId}/items{1}`;
+const apiItem = `https://deliver.kontent.ai/${projectId}/items/{1}`;
 const apiTaxonomies = "https://deliver.kontent.ai/{0}/taxonomies/{1}";
-const itemsParams = "?includeTotalCount={0}&limit={1}&order=elements.{2}";
-const itemsForTagParams = "&elements.blog_tags[contains]={0}";
 const pageSize = 10;
 const tag = getUrlParamater("tag");
 
 const blogSummaryTemplate = document.querySelector("#blog_summary_template");
 const apiErrorMessageTemplate = document.querySelector("#api_error_message");
 const apiNoResultsMessageTemplate = document.querySelector("#api_noresults_message");
 
+class URLBuilder {
+    constructor(baseURL) {
+        this.includeTotalCount = true;
+        this.limit = null;
+        this.orderByElementProp = null;
+        this.blogTagsContains = null;
+        this.baseURL = new URL(baseURL); // so that if the URL isn't legal we know immediately
+    }
+
+    build() {
+        const working = new URL(baseURL);
+        const searchParams = working.searchParams;
+        if (this.includeTotalCount) searchParams.set("includeTotalCount", this.includeTotalCount);
+        if (this.limit) searchParams.set("limit", this.limit);
+        if (this.orderByElementProp) searchParams.set("order", `elements.{${this.orderByElementProp}}`);
+        if (this.blogTagsContains) searchParams.set("elements.blog_tags[contains]", this.blogTagsContains);
+        return working;
+    }
+}
+
+const itemsURLBuilder = new URLBuilder(apiItems);
```

Comment:
Consider renaming - it can also be used for other API urls, so maybe something like kontentApiBuilder

Reply:
yes you are right.